src/sitemap/handler.js

```diff
@@ -395,182 +399,135 @@ export async function sitemapAuditRunner(baseURL, context) {
-        buildKey: (issue) => buildKeyOptionalPage(auditData, issue, 'pages-in-sitemap-have-errors'),
-        title: 'Issues regarding pages in sitemaps',
-        description: 'All pages from sitemap should be available, with no redirects in place.',
-      });
-    }
-  }
+  const { success, reasons } = auditData.auditResult;
+  const response = success
+    ? []
+    : reasons.map(({ error }) => ({ type: 'error', error }));
 
-  return response;
-}
-
-async function createOpportunity(dataAccess, type, title, log, siteId, auditId) {
-  const opportunityData = {
-    siteId,
-    auditId,
-    type,
-    title,
-    runbook: 'https://adobe.sharepoint.com/:w:/r/sites/aemsites-engineering/Shared%20Documents/3%20-%20Experience%20Success/SpaceCat/Runbooks/Experience_Success_Studio_Sitemap_Runbook.docx?d=w6e82533ac43841949e64d73d6809dff3&csf=1&web=1&e=FTWy7t',
-    guidance: {
-      steps: [
-        'Verify each URL in the sitemap, identifying any that do not return a 200 (OK) status code.',
-        'Check RUM data to identify any sitemap pages with unresolved 3xx, 4xx or 5xx status codes – it should be none of them.',
-      ],
-    },
-    tags: ['Traffic Acquisition'],
-  };
-  try {
-    return await dataAccess.Opportunity.create(opportunityData);
-  } catch (e) {
-    log.error(`Failed to create new opportunity for siteId ${siteId} and auditId ${auditId}: ${e.message}`);
-    throw e;
-  }
-}
+  const pagesWithIssues = getPagesWithIssues(auditData);
+  const suggestions = [...response, ...pagesWithIssues].filter(Boolean);
 
-export async function handleClassifiedOpportunity(
-  detectedEntry,
-  existingEntries,
-  log,
-  siteId,
-  auditId,
-  dataAccess,
-) {
-  const newData = detectedEntry.getData();
-  if (newData.length) {
-    let opp = existingEntries.find((oppty) => oppty.getType() === detectedEntry.type);
-    if (opp) {
-      opp = opp.setAuditId(auditId);
-      await opp.save();
-    } else {
-      opp = await createOpportunity(
-        dataAccess,
-        detectedEntry.type,
-        detectedEntry.title,
-        log,
-        siteId,
-        auditId,
-      );
-    }
-
-    await syncSuggestions({
-      opportunity: opp,
-      newData,
-      buildKey: detectedEntry.buildKey,
-      mapNewSuggestion: (issue) => ({
-        opportunityId: opp.getId(),
-        type: 'CODE_CHANGE',
-        rank: issue.traffic_domain ?? 0,
-        data: issue,
-      }),
-      log,
-    });
-  }
+  log.info(`Classified suggestions: ${JSON.stringify(suggestions)}`);
+  return suggestions;
 }
 
 export async function convertToOpportunity(auditUrl, auditData, context) {
   const { dataAccess, log } = context;
 
-  const classifiedOpportunities = await classifyOpportunities(auditUrl, auditData);
-  if (!classifiedOpportunities.length) {
+  log.info('Converting SITEMAP audit to opportunity...');
+
+  const classifiedSuggestions = classifySuggestions(auditUrl, auditData, log);
+  if (!classifiedSuggestions.length) {
```

Comment:
Should we also delete an existing opportunity or remove suggestions if there are no classifiedSuggestions in an audit ?

Reply:
I don't think we are doing that right now for any opportunity. I think that decision should be made on PO level

Reply:
okay.